py/picca/delta_extraction/expected_flux.py

```diff
@@ -112,33 +112,9 @@ def _initialize_variance_wavelength_array(self):
         The initialized arrays are:
         - self.log_lambda_var_func_grid
         """
-        # initialize the variance-related variables (see equation 4 of
-        # du Mas des Bourboux et al. 2020 for details on these variables)
-        if Forest.wave_solution == "log":
-            self.log_lambda_var_func_grid = (
-                Forest.log_lambda_grid[0] +
-                (np.arange(self.num_bins_variance) + .5) *
-                (Forest.log_lambda_grid[-1] - Forest.log_lambda_grid[0]) /
-                self.num_bins_variance)
-        # TODO: this is related with the todo in check the effect of finding
-        # the nearest bin in log_lambda space versus lambda space infunction
-        # find_bins in utils.py. Once we understand that we can remove
-        # the dependence from Forest from here too.
-        elif Forest.wave_solution == "lin":
-            self.log_lambda_var_func_grid = np.log10(
-                10**Forest.log_lambda_grid[0] +
-                (np.arange(self.num_bins_variance) + .5) *
-                (10**Forest.log_lambda_grid[-1] -
-                 10**Forest.log_lambda_grid[0]) / self.num_bins_variance)
-
-        # TODO: Replace the if/else block above by something like the commented
-        # block below. We need to check the impact of doing this on the final
-        # deltas first (eta, var_lss and fudge will be differently sampled).
-        #start of commented block
-        #resize = len(Forest.log_lambda_grid)/self.num_bins_variance
-        #print(resize)
-        #self.log_lambda_var_func_grid = Forest.log_lambda_grid[::int(resize)]
-        #end of commented block
+        resize = len(Forest.log_lambda_grid)/self.num_bins_variance
+        self.log_lambda_var_func_grid = Forest.log_lambda_grid[::int(resize)]
```

Comment:
wouldn't that be assymetric? I.e.. we're discarding on the high-lambda end always? I guess that's fine given that most observations are at the low-redshift end.
But wouldn't we actually want something like `np.logspace(np.min(Forest.log_lambda_grid),np.max(Forest.log_lambda_grid),self.num_bins_variance [maybe +1?])`?

Also the previous version had the variance bins defined to be bin centers, would we now define them as bin edges? Given that the lowest wavelength point is always included? Maybe that is the correct way  now, given that the find_bins function used to select the closest entries and now selects entries that have higher lambda only (is this actually what it does now?).